crypto/test/src/crypto/test/OpenSSHKeyParsingTests.cs

```diff
@@ -0,0 +1,521 @@
+                "OMx2MFIx4v0tJZvQvyiUxl5IJJ9ZJsYUWF+6VbcTVwYYfdVzZzP2TNyGmF9/ADZW\n" +
+                "wBehqP04uRlYjt94kqb4HoOKF3gJ3LC4uW9xcEltTBeHWCfhAkEA/2biF5St9/Ya\n" +
+                "540E4zu/FKPsxLSaT8LWCo9+X7IqIzlBQCB4GjM+nZeTm7eZOkfAFZoxwfiNde/9\n" +
+                "qleXXf6B2QJBAPAW+jDBC3QF4/g8n9cDxm/A3ICmcOFSychLSrydk9ZyRPbTRyQC\n" +
+                "YlC2mf/pCrO/yO7h189BXyQ3PXOEhnujce8CQQD7gDy0K90EiH0F94AQpA0OLj5B\n" +
+                "lfc/BAXycEtpwPBtrzvqAg9C/aNzXIgmly10jqNAoo7NDA2BTcrlq0uLa8xBAkBl\n" +
+                "7Hs+I1XnZXDIO4Rn1VRysN9rRj15ipnbDAuoUwUl7tDUMBFteg2e0kZCW/6NHIgC\n" +
+                "0aG6fLgVOdY+qi4lYtfFAkEAqqiBgEgSrDmnJLTm6j/Pv1mBA6b9bJbjOqomrDtr\n" +
+                "AWTXe+/kSCv/jYYdpNA/tDgAwEmtkWWEie6+SwJB5cXXqg==\n" +
+                "-----END RSA PRIVATE KEY-----\n")).ReadPemObject().Content);
+
+            DoRSATest(pubSpec, privSpec);
+
+            privSpec = OpenSSHPrivateKeyUtil.ParsePrivateKeyBlob(new PemReader(new StringReader(rsa1024Key)).ReadPemObject().Content);
+            pubSpec = new RsaKeyParameters(false, ((RsaKeyParameters)privSpec).Modulus, ((RsaPrivateCrtKeyParameters)privSpec).PublicExponent);
+
+            DoRSATest(pubSpec, privSpec);
+
+            privSpec = OpenSSHPrivateKeyUtil.ParsePrivateKeyBlob(new PemReader(new StringReader(rsa2048Key)).ReadPemObject().Content);
+            pubSpec = new RsaKeyParameters(false, ((RsaKeyParameters)privSpec).Modulus, ((RsaPrivateCrtKeyParameters)privSpec).PublicExponent);
+
+            DoRSATest(pubSpec, privSpec);
+
+            privSpec = OpenSSHPrivateKeyUtil.ParsePrivateKeyBlob(new PemReader(new StringReader(rsa3072Key)).ReadPemObject().Content);
+            pubSpec = new RsaKeyParameters(false, ((RsaKeyParameters)privSpec).Modulus, ((RsaPrivateCrtKeyParameters)privSpec).PublicExponent);
+
+            DoRSATest(pubSpec, privSpec);
+
+            privSpec = OpenSSHPrivateKeyUtil.ParsePrivateKeyBlob(new PemReader(new StringReader(rsa4096Key)).ReadPemObject().Content);
+            pubSpec = new RsaKeyParameters(false, ((RsaKeyParameters)privSpec).Modulus, ((RsaPrivateCrtKeyParameters)privSpec).PublicExponent);
+
+            DoRSATest(pubSpec, privSpec);
+        }
+
+        private void DoRSATest(AsymmetricKeyParameter pubSpec, AsymmetricKeyParameter privSpec)
+        {
+            byte[] originalMessage = new byte[10];
+            secureRandom.NextBytes(originalMessage);
+
+            originalMessage[0] |= 1;
+
+            var rsaEngine = new RsaEngine();
+            rsaEngine.Init(true, privSpec);
+
+            byte[] ct = rsaEngine.ProcessBlock(originalMessage, 0, originalMessage.Length);
+
+            rsaEngine.Init(false, pubSpec);
+            byte[] result = rsaEngine.ProcessBlock(ct, 0, ct.Length);
+
+            IsTrue("Result did not match original message", Enumerable.SequenceEqual(originalMessage, result));
+        }
+
+        [Test]
+        public void TestFunction()
```

Comment:
Is this an artifact from testing?

Reply:
> Is this an artifact from test setup?

All the tests in BouncyCastle.Crypto.Tests have this function.